File: ChatSwift/ChatSwift/viewmodel/SohbetListesiViewModel.swift
import Foundation
import Combine

final class SohbetListesiViewModel {
    
    @Published private(set) var konusulanKisiler: [KonusulanKisi] = []
    @Published private(set) var hataMesaji: String?
    
    private let api = ApiService.shared
    private var yenilemeTimer: AnyCancellable?
    
    /// 15 saniyede bir konuşulan kişileri getirir
    func sohbetListesiniBaslat(kullaniciId: Int) {
        yenilemeTimer?.cancel()
        sohbetListesiniGetir(kullaniciId: kullaniciId) // hemen ilk yükleme
        
        yenilemeTimer = Timer.publish(every: 15, on: .main, in: .common)
            .autoconnect()
            .sink { [weak self] _ in
                self?.sohbetListesiniGetir(kullaniciId: kullaniciId)
            }
    }

    private func sohbetListesiniGetir(kullaniciId: Int) {
        api.konusulanKisiler(kullaniciId: kullaniciId) { [weak self] result in
            switch result {
            case .success(let response):
                if response.success {
                    self?.konusulanKisiler = response.kisiler
                    self?.hataMesaji = nil
                } else {
                    self?.hataMesaji = "Liste alınamadı"
                }
            case .failure(let error):
                self?.hataMesaji = "Sunucu hatası: \(error.localizedDescription)"
            }
        }
    }

    deinit {
        yenilemeTimer?.cancel()
    }
}

File: flutter_chat_uygulama/lib/viewmodel/sohbet_listesi_viewmodel.dart
import 'dart:async';

import 'package:flutter/material.dart';

import '../model/konusulan_kisi.dart';
import '../service/api_service.dart';

class SohbetListesiViewModel extends ChangeNotifier {
  final ApiService _api;

  SohbetListesiViewModel({ApiService? api}) : _api = api ?? ApiService();

  List<KonusulanKisi> _konusulanKisiler = [];
  List<KonusulanKisi> get konusulanKisiler => _konusulanKisiler;

  String? _hataMesaji;
  String? get hataMesaji => _hataMesaji;

  Timer? _yenilemeTimer;

  /// Kotlin: sohbetListesiniBaslat(kullaniciId)
  void sohbetListesiniBaslat({required int kullaniciId}) {
    _yenilemeTimer?.cancel();

    // İlk yükleme hemen
    _fetch(kullaniciId);

    _yenilemeTimer = Timer.periodic(const Duration(seconds: 15), (_) {
      _fetch(kullaniciId);
    });
  }

  Future<void> _fetch(int kullaniciId) async {
    try {
      final response = await _api.konusulanKisiler(kullaniciId: kullaniciId);

      if (response.success) {
        _konusulanKisiler = response.kisiler;
        _hataMesaji = null;
      } else {
        _hataMesaji = 'Liste alınamadı';
      }
    } catch (e) {
      _hataMesaji = 'Sunucu hatası: $e';
    }
    notifyListeners();
  }

  void stop() {
    _yenilemeTimer?.cancel();
    _yenilemeTimer = null;
  }

  @override
  void dispose() {
    stop();
    super.dispose();
  }
}
